src/main/java/roomescape/service/PaymentService.java

```diff
@@ -0,0 +1,43 @@
+package roomescape.service;
+
+import org.springframework.http.HttpStatusCode;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestClient;
+import roomescape.controller.request.PaymentRequest;
+import roomescape.controller.request.ReservationRequest;
+import roomescape.exception.PaymentException;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+
+@Service
+public class PaymentService {
+
+    private final static long RESERVATION_PRICE = 1999999;
+
+    private final RestClient restClient;
+
+    public PaymentService(RestClient restClient) {
+        this.restClient = restClient;
+    }
+
+    public void confirmReservationPayments(ReservationRequest request) {
+        validatePayments(request.amount());
+        restClient.post()
+                .uri("/confirm")
+                .contentType(APPLICATION_JSON)
+                .body(new PaymentRequest(request.paymentKey(), request.orderId(), request.amount()))
+                .retrieve()
+                .onStatus(HttpStatusCode::is4xxClientError, (req, res) -> {
+                    throw new PaymentException("결제 정보가 일치하지 않습니다.");
+                })
+                .onStatus(HttpStatusCode::is5xxServerError, (req, res) -> {
+                    throw new PaymentException("결제 시스템이 원활하게 동작하지 않습니다.");
+                }).toBodilessEntity();
```

Comment:
```suggestion
                .onStatus(HttpStatusCode::is4xxClientError, (req, res) -> {
                    throw new PaymentException("결제 정보가 일치하지 않습니다.");
                })
                .onStatus(HttpStatusCode::is5xxServerError, (req, res) -> {
                    throw new PaymentException("결제 시스템이 원활하게 동작하지 않습니다.");
                })
                .toBodilessEntity();
```
민감한 정보에 대해서 적절히 추상화 해주셨네요 👍
하나 더 생각해볼 부분은 만약 외부 API에서 문제가 발생해 수십초간 응답이 오지 않는다면 어떻게 될까요?
Connection Timeout과 Read Timeout은 어떤 차이가 있나요?